pandas/core/arrays/base.py

```diff
@@ -0,0 +1,203 @@
+"""An interface for extending pandas with custom arrays."""
```

Comment:
I'd advocate leaving base.py open for (near-)future usage as pandas-internal base and putting the "use this if you want to write your own" file in e.g. extension.py

Reply:
I have a slight preference for `base.py` since it's a base class for all extension arrays. I don't think that having `ExtensionArray` in `arrays.base`  precludes having a pandas-internal base there as well.

Reply:
It will be publicly exposed through pd.api.extensions anyway I think

Reply:
Adding stuff to the public API is waiting on https://github.com/pandas-dev/pandas/pull/19304